core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/pipeline/stages/TimeoutExceptionHandlingStage.java

```diff
@@ -90,7 +91,8 @@ public TimeoutExceptionHandlingStage(HttpClientDependencies dependencies, Reques
      */
     private Exception translatePipelineException(RequestExecutionContext context, Exception e) {
         if (e instanceof InterruptedException || e instanceof IOException ||
-            e instanceof AbortedException || Thread.currentThread().isInterrupted()) {
+                e instanceof AbortedException || Thread.currentThread().isInterrupted()
+                || (e instanceof SdkClientException && isCausedByApiCallAttemptTimeout(context))) {
```

Comment:
We check `isCausedByApiCallAttemptTimeout` but can this be caused by the API call timer as well?

Reply:
The timeout exception is caused while  [ApiCallAttemptTimeoutExecute](https://github.com/aws/aws-sdk-java-v2/blob/35267ca707c3fb5cdf7e3e98758a8ef969269183/core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/pipeline/stages/ApiCallAttemptTimeoutTrackingStage.java#L73) and this is the same check which resets the interrupts 
[ApiCallAttemptTimeoutTrackingStageInterruptReset](https://github.com/aws/aws-sdk-java-v2/blob/35267ca707c3fb5cdf7e3e98758a8ef969269183/core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/pipeline/stages/ApiCallAttemptTimeoutTrackingStage.java#L135).

API call timer  causes context.apiCallTimeoutTracker() while the one used above is 
context.apiCallAttemptTimeoutTracker()

Reply:
Okay gotcha, so if the API call timer fired (and _not_ the attempt timer), then the exception would be `AbortedException` which we already handle.